src/caselawclient/Client.py

```diff
@@ -448,23 +480,7 @@ def invoke(
         self._raise_for_status(response)
         return response
 
-    def advanced_search(
-        self,
-        q: Optional[str] = None,
-        court: Optional[str] = None,
-        judge: Optional[str] = None,
-        party: Optional[str] = None,
-        neutral_citation: Optional[str] = None,
-        specific_keyword: Optional[str] = None,
-        order: Optional[str] = None,
-        date_from: Optional[str] = None,
-        date_to: Optional[str] = None,
-        page: int = 1,
-        page_size: int = RESULTS_PER_PAGE,
-        show_unpublished: bool = False,
-        only_unpublished: bool = False,
-        collections: Optional[str] = None,
-    ) -> requests.Response:
+    def advanced_search(self, search_parameters: SearchParameters) -> requests.Response:
```

Comment:
just take in a SearchParameters object now for all the search functions in the search chain as explained on the comment at the SearchParameters definition